ogr/ogr_geometry.h

```diff
@@ -307,6 +308,134 @@ class CPL_DLL OGRDefaultConstGeometryVisitor : public IOGRConstGeometryVisitor
+    int nXYBitPrecision =
+        INT_MIN; /**< Number of bits needed to achieved XY precision. Typically
+                    computed with OGRPrecisionOptionsSetPrecision() or
+                    OGRPrecisionOptionsSetMetricPrecision() */
+    int nZBitPrecision =
+        INT_MIN; /**< Number of bits needed to achieved Z precision. Typically
+                    computed with OGRPrecisionOptionsSetPrecision() or
+                    OGRPrecisionOptionsSetMetricPrecision() */
+    int nMBitPrecision =
+        INT_MIN; /**< Number of bits needed to achieved M precision. Typically
+                    computed with OGRPrecisionOptionsSetPrecision() or
+                    OGRPrecisionOptionsSetMetricPrecision() */
+};
+
+/************************************************************************/
+/*                           OGRwkbExportOptions                        */
+/************************************************************************/
+
+/** WKB export options.
+ *
+ * @since GDAL 3.7
+ */
+struct OGRwkbExportOptions
+{
+    OGRwkbByteOrder eByteOrder = wkbNDR;          /**< Byte order */
+    OGRwkbVariant eWkbVariant = wkbVariantOldOgc; /**< WKB variant. */
+    OGRPrecisionOptions sPrecision{};             /**< Precision options. */
+};
+
+/************************************************************************/
+/*                        OGRQuantizeValue()                            */
+/************************************************************************/
+
+/** Set to zero least significants bits of a double precision floating-point
+ * number (passed as an integer), taking into account a desired bit precision.
+ *
+ * @param nVal Integer representation of a IEEE754 double-precision number.
+ * @param nBitsPrecision Desired precision (number of bits after integral part)
+ * @return quantized nVal.
+ * @since GDAL 3.7
+ */
+inline uint64_t OGRQuantizeValue(uint64_t nVal,
+                                 int nBitsPrecision) CPL_WARN_UNUSED_RESULT;
+
+inline uint64_t OGRQuantizeValue(uint64_t nVal, int nBitsPrecision)
+{
+    // Extact the binary exponent from the IEEE754 representation
+    const int nExponent = ((nVal >> 52) & 2047) - 1023;
+    // Add 1 to round-up and the desired precision
+    const int nBitsRequired = 1 + nExponent + nBitsPrecision;
+    // Compute number of nullified bits
+    int nNullifiedBits = 52 - nBitsRequired;
+    // this will also capture NaN and Inf since nExponent = 1023,
+    // and thus nNullifiedBits < 0
+    if (nNullifiedBits <= 0)
+        return nVal;
+    if (nNullifiedBits >= 52)
+        nNullifiedBits = 52;
+    nVal &= (~uint64_t(0)) << nNullifiedBits;
+    return nVal;
+}
+
+/************************************************************************/
+/*                        OGRQuantizeXYValues()                         */
+/************************************************************************/
+
+/** Quantize XY values.
+ *
+ * @since GDAL 3.7
+ */
+template <int SPACING>
```

Comment:
Why a template and not a simple argument?

Reply:
To generate better assembly code.   ptr + variable * constant can be expressed into a single assembly mnemonic